wallet-new/docs/updates.md

```diff
@@ -497,179 +520,310 @@ The way updates will transition from the *next* to the *current* can be the foll
+      description: >
+        Vote with stake on the update proposal. Voting must
+        abide to the [votes verification rules](https://github.com/input-output-hk/cardano-sl/blob/master/docs/block-processing/us.md#votes-verification)
+        and issuing the same vote twice, or changing the vote more than one
+        time is not allowed.
+      parameters:
+        - in: path
+          name: proposalId
+          type: string
+          required: true
+          description: Proposal ID to vote on
+        - name: body
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/UpdateProposalVote'
+      consumes:
+        - application/json;charset=utf-8
+      responses:
+        '204':
+          description: 'Vote has been succesful.'
+        '403':
+          description: 'Invalid stake key.'
+        '404':
+          description: 'Voted proposal is not active anymore.'
+        '400':
+          description: 'Not enough stake to vote on this proposal.'
+definitions:
+  Installer:
+    description: "A Daedalus installer."
+    type: object
+    properties:
+      platform:
         type: string
-        enum: [mandatory, optional]
-      UpdateState:
-      # Just a stub for now.
-        type: object
+        example: "linux"
+      url:
+        type: string
+        example: "https://s3.iohk.io/installers/daedalus.zip"
+      checksum:
+        description: "A Blake2b-256 hash."
+        example: "A8ADD4BDDDFD93E4877D2746E62817B116364A1FA7BC148D95090BC7333B3673"
+        type: string
+  ProtocolUpgrade:
+    type: object
+    properties:
+      maxHeaderSize:
+        type: integer
+        maximum: 65535
+        minimum: 0
+  Installers:
+    type: array
+    items:
+      $ref: '#/definitions/Installer'
+  UpdateProposal:
+    type: object
+    description:
+      An update proposal.
+    properties:
+      blockVersion:
+         $ref: '#/definitions/BlockVersion'
+      protocolUpgrade:
+         $ref: '#/definitions/ProtocolUpgrade'
+      installers:
+         $ref: '#/definitions/Installers'
+      softwareVersion:
+         $ref: '#/definitions/SoftwareVersion'
+    required:
+       - blockVersion
+       - protocolUpgrade
+       - installers
+       - softwareVersion
+  ProposedUpdate:
+    type: object
+    description: >
+      Bundles an `UpdateId` and an `UpdateProposal` together.
+    properties:
+      proposalId:
+          type: string
+          description: "The hash of the UpdateProposal (Blake2b_256)."
+          example: "A8ADD4BDDDFD93E4877D2746E62817B116"
+      updateProposal:
+        $ref: '#/definitions/UpdateProposal'
+    required:
+      - proposalId
+  UpdateProposalVote:
+    type: object
+    properties:
+      vote:
+        $ref: '#/definitions/Vote'
+      stakeKey:
+        type: string
+        description: "The delegation stake key for this node."
+        example: "A8ADD4BDDDFD93E4877D2746E62817B116364A1FA7BC148D95090BC7333B3673"
```

Comment:
It should not be a stake key, but rather a reference. Also, this would be useful only if user has multiple stake keys on his node available it it's the case.

I would say it's questionable whether we want it done this way (explicitly pass stake key). My initial complaint was more about you have to define which stake key would be used. I.e. show user that he send vote with this stake key (b/c this one is used by his node for operations).

Reply:
BTW, Update proposal also requires stake key to be created